analyzer/src/main/kotlin/Analyzer.kt

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (C) 2017-2018 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package com.here.ort.analyzer
+
+import ch.frankel.slf4k.*
+
+import com.here.ort.analyzer.managers.Unmanaged
+import com.here.ort.downloader.VersionControlSystem
+import com.here.ort.model.AnalyzerConfiguration
+import com.here.ort.model.AnalyzerResultBuilder
+import com.here.ort.model.AnalyzerRun
+import com.here.ort.model.Environment
+import com.here.ort.model.OrtResult
+import com.here.ort.model.ProjectAnalyzerResult
+import com.here.ort.model.Repository
+import com.here.ort.model.config.RepositoryConfiguration
+import com.here.ort.utils.log
+
+import java.io.File
+
+class Analyzer {
+    fun analyze(config: AnalyzerConfiguration, absoluteProjectPath: File,
+                packageManagers: List<PackageManagerFactory<PackageManager>> = PackageManager.ALL,
+                packageCurationsFile: File? = null
+    ): OrtResult {
+        // Map of files managed by the respective package manager.
+        val managedDefinitionFiles = if (packageManagers.size == 1 && absoluteProjectPath.isFile) {
+            // If only one package manager is activated, treat the given path as definition file for that package
+            // manager despite its name.
```

Comment:
How about "regardless of its name"?

Reply:
This code was just moved to another file, I would address this in a separate PR.